### PR TITLE
[BugFix] fix problems in data split

### PR DESCRIPTION
**Pull request by kayzliu**
## Description

1. fix ineffective permutation.
2. fix improper index arrangement, which can result in incorrect splitting in Amazon dataset.
3. fix format problems in doc-string
4. fix statistics
@zhjwy9343 

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change

### CODE REVIEW

File: python/dgl/data/fraud.py [hunk 1]
```diff
@@ -213,13 +213,16 @@ class FraudYelpDataset(FraudDataset):
 
     - Nodes: 45,954
     - Edges:
-        R-U-R: 49,315
-        R-T-R: 573,616
-        R-S-R: 3,402,743
-        ALL: 3,846,979
+
+        - R-U-R: 98,630
```
Comment (zhjwy9343) [inline python/dgl/data/fraud.py:217]: What happened here? Did you add reversed edges?

File: python/dgl/data/fraud.py [hunk 2]
```diff
@@ -269,23 +272,27 @@ class FraudAmazonDataset(FraudDataset):
...
     Statistics:
 
     - Nodes: 11,944
     - Edges:
-        U-P-U: 175,608
-        U-S-U: 3,566,479
-        U-V-U: 1,036,737
-        ALL: 4,398,392
+
+        - U-P-U: 351,216
```
Comment (zhjwy9343) [inline python/dgl/data/fraud.py:285]: ditto

File: python/dgl/data/fraud.py (hunk 1 above)
Comment (kayzliu) [reply python/dgl/data/fraud.py:217]: > What happened here? Did you add reversed edges?

As the dostring in line 25 described, the original dataset is bidirectional. However, the statistics from original paper hasn't take reversed edges in to account. Therefore, the number of edges should be doubled.

File: python/dgl/data/fraud.py (hunk 2 above)
Comment (kayzliu) [reply python/dgl/data/fraud.py:285]: > ditto

ditto

**Review by zhjwy9343 — APPROVED** [review]
LGTM

**Pull request merged**